### ISSUE TITLE
Support for WPML plugin

### ISSUE THREAD

**Issue by raykai** [soflyy/oxygen-bugs-and-features#938]
**Describe the feature you'd like to see included in Oxygen.**
*A clear, detailed description of the feature you're requesting.*
Support for WPML [https://wpml.org](https://wpml.org) plugin would be good to have for Oxygen

**What are the use cases for this feature?**
*e.g. Accessibility, e-commerce store that sells bicycles, etc.*
Language Accessibility

**Examples of this feature or functionality.**
*If another product has done it well, provide examples here.*
 It would allow users to use more than one language on their sites when they build using Oxygen. 
I live in a country where we need two languages on our sites so having support for WPML is almost a must for us.

**Comment by KarzanTT** [comment]
+1 Support Oxygen for WPML plugin!

**Comment by Xilonz** [comment]
I've used WPML on an Oxygen site and it works pretty well, I'd say it's usable. If you rely on translators directly translating in WPML, Oxygen won't work for you.

### What works: 
- Routing in general
- Duplicating posts to another language
- Templates based on language (I've set them to: `Translatable - use translation if available or fallback to default language`) 

### What doesn't work
- Advanced translation editor. From my understanding, it should be possible with just a config file if the Oxygen team puts time and effort into it.
- Reusable template translations don't get picked up inside templates or blocks, you have to replace them everywhere you use them. This should be really easy to fix. 
- Link URLs don't get translated to their language equivalent.

### Buggy behavior:
- Sometimes when you edit a template you end up editing the wrong language (I believe this only happens if you use the `?lang=` structure in WPML.

### Tip:
Use Oxygen Gutenberg blocks, they can easily be cloned to the other language and edited. Giving you enough power to barely need to translate templates or navigate to the Oxygen builder.

**Comment by Xilonz** [comment]
WPML Oxygen Caching support for duplicated pages;
cache.php:297
```php
       ...
	// finally enqueue styles
	foreach ($oxygen_vsb_css_files_to_load as $post_id) {

            $post_id = apply_filters( 'wpml_master_post_from_duplicate', $post_id ) ?: $post_id;
        ...
```

**Comment by Xilonz** [comment]
In order to see translated templates when not logged in, modify `/component-framework/admin/cpt-template.php`. Please note that this will make the templates publicly_queryable. 
```diff
...
function ct_add_templates_cpt() {

-	if(!oxygen_vsb_current_user_can_access()) {
-		return;
-	}

	$labels = array(
		'name'               => _x( 'Templates', 'post type general name', 'component-theme' ),
		'singular_name'      => _x( 'Template', 'post type singular name', 'component-theme' ),
		'menu_name'          => _x( 'Templates', 'admin menu', 'component-theme' ),
		'name_admin_bar'     => _x( 'Template', 'add new on admin bar', 'component-theme' ),
		'add_new'            => _x( 'Add New Template', 'template', 'component-theme' ),
		'add_new_item'       => __( 'Add New Template', 'component-theme' ),
		'new_item'           => __( 'New Template', 'component-theme' ),
		'edit_item'          => __( 'Edit Template', 'component-theme' ),
		'view_item'          => __( 'View Template', 'component-theme' ),
		'all_items'          => __( 'Templates', 'component-theme' ),
		'search_items'       => __( 'Search Templates', 'component-theme' ),
		'parent_item_colon'  => __( 'Parent Templates:', 'component-theme' ),
		'not_found'          => __( 'No templates found.', 'component-theme' ),
		'not_found_in_trash' => __( 'No templates found in Trash.', 'component-theme' )
	);
       ...
```

**Comment by Netzlichter** [comment]
Please make oxygen wpml compatible. Multilingual is so important...

**Comment by out-of-order** [comment]
very important!

**Comment by KarzanTT** [comment]
UpVote, Very important to make oxygen works with WPML plugin.

**Comment by out-of-order** [comment]
How can i upvote?

**Comment by out-of-order** [comment]
It is a fundamental connection between one of the best construction kits and the IMHO best maintained Multilanguage building block.

**Comment by out-of-order** [comment]
can anyone help with the second language search page results template to pick up the duplicated main template? If I set the priority, then it also picked up in the first language.

Nearly nothings works. I had to duplicate all pages page per page.
The management tool sucked totally.
Some text fragments are displayed in the translation tool, but not displayed in the frontend.

It sucks hard.
It is absolutely the worst and most imaginable trashy thing.
I just hate to work with it!

**Comment by Netzlichter** [comment]
Don't use oxygen with wpml for client or important production sites. I think it is the only option at the moment. If you know find a hacky solution it is not save for the future... I would not use it at the moment.

**Comment by out-of-order** [comment]
But I need a way to make it multilang!

**Comment by out-of-order** [comment]
Thanks to Kevin from Oxygen I was able to solve my task.
To translate a page:

Pages
- click on the page to edit
- click on duplicate and choose copy media
- edit the slug
- sync the menu at wpml settings after you duplicated all pages.
- change to new language in WP backend and edit the new pages with Oxygenbuilder

Posts (Gutenberg)
- that's easy. You can directly translate it with WPML. I have used the classic tool. The new sucks IMHO.

- Posts (Oxygen)
 Do so like you would do with Pages

Don't use the duplicate plugin to copy templates. Again here the adverts of WPML are wrong. That doesn't work at all.
Only main and Pages do work this way. But what with 404, Posts, Search etc...?

I lost so much time with it. Her you go with templates:

To get elements like fake breadcrumbs or footer elements (and all other individual stuff that brings you freedom to do as you want to) use conditions:

1. install the Code Snippet. This snippet checks for a language string within the URL when the function check_language_url is called. For example, /en/ or /fr/, and then returns a string for the language. In the future, you may add additional languages as you wish.

`<?php

/**
 * Check Language URL
 */
function check_language_url() {
	
	$url = $_SERVER['REQUEST_URI'];

	$language = "deutsch";
	
    if (strpos($url, '/en/') !== false) {
        $language = "english";
    } elseif (strpos($url, '/fr/') !== false) {
		$language = "french";
	}
	
	return $language;
	
}`

2. you may go into any one of your Oxygen templates and copy any elements you wish to translate and translate them within the editor

3. Finally, to set up the condition settings ():

    Select an element that you would like to show for a language
    Click the Condition Settings icon
    Click Set Conditions > Dynamic Data > PHP Function Return Value
    Enter "check_language_url" into the Function Name (without quotes)
    Enter the return value of the language you would like to display it for.
    You will need to set this up for all elements you would like to hide or show based on a language barrier.

**Comment by MDGDesign** [comment]
Voting for full compatibility ASAP with WPML.

I am from the DIVI ( Bloated) world and am looking forward to have a good page builder like Oxygen compatible with WPML.

**For does looking for some help on the subject:**
- https://wpml.org/forums/topic/oxygen-builder-translation/#post-8434339
- https://wpml.org/documentation/plugins-compatibility/oxygen-visual-builder/#translating-oxygen-reusable-elements-templates
- 
For WPML config file to get some fields into translation:

`<wpml-config>`
 `<shortcodes>`
    `<shortcode>`
     `  <tag>oxy_rich_text</tag>`
  `  </shortcode>`
   ` <shortcode>`
         ` <tag>ct_link_text</tag>`
    `</shortcode>`
   ` <shortcode>`
         ` <tag>ct_link_button</tag>`
  `  </shortcode>`
` </shortcodes>`
`</wpml-config>`

IF any body knows how to get the link to translation in any modules with links that would help!
Maybe a attribute tag?

Thanks!

**Comment by raykai** [comment]
+1 for WPML support!

**Comment by nil80** [comment]
+1 for WPML support please!

**Comment by Nemesis19** [comment]
+1

**Comment by danpii** [comment]
+1

**Comment by octapress** [comment]
+1

**Comment by JonathanF88** [comment]
+1 for WPML. If you are serious enough, we need WPML support! Templates would be a nice starting point.

**Comment by kartzke** [comment]
+1 for WPML support!

**Comment by angang** [comment]
Native support without tricks is vert very important.

**Comment by Schokolader** [comment]
+1

**Comment by knut-studio** [comment]
+1

**Comment by Jehu** [comment]
We really need a solution so we can use WPML's translation Editor instead of using this workaround mentioned on https://wpml.org/errata/oxygen-builder-widget-and-link-translation-issues/
Can't get https://wpml.org/forums/topic/oxygen-builder-translation/#post-8600795 to work, but this can't be impossible to solve for the Oxygen/WPML devs.

**Comment by strategio** [comment]
+1

**Comment by ofwald** [comment]
+1

**Comment by dgwatkins** [comment]
+1 for WPML support!

**Comment by Provision1080** [comment]
+1! Support Oxygen for WPML plugin!!

**Comment by parodev** [comment]
Yes, please +1

**Comment by kurosato** [comment]
+1, please

**Comment by montrealist** [comment]
Yess, WPML support for the win!

**Comment by SinjoriS** [comment]
+1 for fully WPML support!

**Comment by szymonsd** [comment]
+1 for WPML support!

**Comment by adrien-robert** [comment]
+1

**Comment by Uxksu** [comment]
🆙Vote, Very important to make oxygen works with WPML plugin.

**Comment by SinjoriS** [comment]
+1 for WPML support! 
We are in 2022 - and no WPML fully support. That's a bad joke!

**Comment by madjedo** [comment]
An interesting behavior/bug:

Try this on a staging site, on a translated page:

Select an element,
Delete all styles from the id,
Save page,
Repeat, this time on same page, default language.

Does the translated version of the page revert back all its content to the default language?

**Comment by SinjoriS** [comment]
> An interesting behavior/bug:
> Try this on a staging site, on a translated page:
> Select an element, Delete all styles from the id, Save page, Repeat, this time on same page, default language.
> Does the translated version of the page revert back all its content to the default language?

Yes, that's tue. I can confirm also on my side. Just tested on one of my Site, and bug appears.

**Comment by Jehu** [comment]
@madjedo interesting! We had the revert bug on some pages in a big project and could find the reason nor replicate it. But your finding is a valuable hint.

**Comment by SinjoriS** [comment]
> An interesting behavior/bug:
> Try this on a staging site, on a translated page:
> Select an element, Delete all styles from the id, Save page, Repeat, this time on same page, default language.
> Does the translated version of the page revert back all its content to the default language?

In fact, that bug happens when you change styling by Oxygen (font size, or color on text), on default language - on the page. When you hit Save button after that - translation is lost on the page with additional language. 
The same bug appears when you change for instance the link on the button - on default language page.

**Comment by Jehu** [comment]
@SinjoriS thats horrible!

**Comment by d-packs** [comment]
+1 WPML support is buggy. I'm at a loss here. 
For some odd reason the translated template for Single product is correctly picked up only when logged in. When not logged in the default language of the template is always used, even if the content itself is translated (meaning the language is correctly detected).

**Comment by madjedo** [comment]
@d-packs That workaround is how I translate my products, it works ok except a few quirks.

But interesting he mentions they are in contact with Oxygen team for this. @Spellhammer Are you perhaps in stealth working on a official integration? I know your policy is not too reveal any future work, but this one is important for a lot of businesses and their future direction on using Oxygen builder and WPML, especially for e-commerce sites.

**Comment by steverevere** [comment]
+1. Need this one big time. 

Thanks!

**Comment by SinjoriS** [comment]
> An interesting behavior/bug:
> 
> Try this on a staging site, on a translated page:
> 
> Select an element, Delete all styles from the id, Save page, Repeat, this time on same page, default language.
> 
> Does the translated version of the page revert back all its content to the default language?

Regarding this bug, here it is solution (answer) from WPML official support. They asked Oxygen support to check that. I hope somebody from Oxygen team will see this. I also sent them a ticked on their Site. 

Open this file: wp-content\plugins\oxygen\component-framework\includes\wpml-support.php around the line 19 you will see the code:
___________________________

add_filter( 'wpml_pb_shortcodes_save_translation', 'ct_wpml_filter_save_translation', 10, 3 );
function ct_wpml_filter_save_translation( $saved, $translated_post_id, $new_content ) {
update_post_meta( $translated_post_id, "ct_builder_shortcodes", $new_content );
return true;
}
___________________________

Please, comment this line of code:
//add_filter( 'wpml_pb_shortcodes_save_translation', 'ct_wpml_filter_save_translation', 10, 3 );

**Comment by 675056817** [comment]
+1

**Comment by marcorubiol** [comment]
We need that

**Comment by spinoWeb2** [comment]
+1

**Comment by d-packs** [comment]
> In order to see translated templates when not logged in, modify `/component-framework/admin/cpt-template.php`. Please note that this will make the templates publicly_queryable.
> 
> ```diff
> ...
> function ct_add_templates_cpt() {
> 
> -	if(!oxygen_vsb_current_user_can_access()) {
> -		return;
> -	}
> 
> 	$labels = array(
> 		'name'               => _x( 'Templates', 'post type general name', 'component-theme' ),
>        ...
> ```

Instead of removing that block, you can move it two blocks lower, after the $args definition and replace its content.
But even better you don't touch Oxygen's code because you lose the changes on update.

luckily `tera_ct_add_templates_cpt()` is being called via a hook. so we just need to unhook it and hook our modified function instead. So simply paste this code in your own plugin or via your own method of adding custom code to Oxygen:

 ```
add_action( 'plugins_loaded', 'tera_oxy_loaded' );
function tera_oxy_loaded() {
	remove_action( 'init', 'ct_add_templates_cpt' );
	add_action( 'init', 'tera_ct_add_templates_cpt' );
}

function tera_ct_add_templates_cpt() {

	$labels = array(
		'name'               => _x( 'Templates', 'post type general name', 'component-theme' ),
		'singular_name'      => _x( 'Template', 'post type singular name', 'component-theme' ),
		'menu_name'          => _x( 'Templates', 'admin menu', 'component-theme' ),
		'name_admin_bar'     => _x( 'Template', 'add new on admin bar', 'component-theme' ),
		'add_new'            => _x( 'Add New Template', 'template', 'component-theme' ),
		'add_new_item'       => __( 'Add New Template', 'component-theme' ),
		'new_item'           => __( 'New Template', 'component-theme' ),
		'edit_item'          => __( 'Edit Template', 'component-theme' ),
		'view_item'          => __( 'View Template', 'component-theme' ),
		'all_items'          => __( 'Templates', 'component-theme' ),
		'search_items'       => __( 'Search Templates', 'component-theme' ),
		'parent_item_colon'  => __( 'Parent Templates:', 'component-theme' ),
		'not_found'          => __( 'No templates found.', 'component-theme' ),
		'not_found_in_trash' => __( 'No templates found in Trash.', 'component-theme' )
	);

	$args = array(
		'exclude_from_search' => true,
		'labels'              => $labels,
		'public'              => true,
		'publicly_queryable'  => true,
		'has_archive'         => true,
		'show_ui'             => true,
		'show_in_menu'        => 'ct_template',
		'rewrite'             => false,
		'query_var'           => true,
		'capability_type'     => 'post',
		'hierarchical'        => false,
		'menu_position'       => null,
		'supports'            => array( 'title' )
	);

	if ( ! oxygen_vsb_current_user_can_access() ) {
		$args['has_archive']        = false;
		$args['query_var']          = false;
		$args['publicly_queryable'] = false;
	}

	register_post_type( 'ct_template', $args );

	// flush rewrite rules if needed
	$flag = get_option( "oxygen_rewrite_rules_updated" );
	if ( $flag !== "1" ) {
		flush_rewrite_rules();
		update_option( "oxygen_rewrite_rules_updated", "1" );
	}
}
 ```

**Comment by zjoko** [comment]
+1

**Comment by d-packs** [comment]
@zjoko the code snippet above makes the templates translatable and it works well with WPML.

**Comment by JamboMedia** [comment]
+1

**Comment by aleknovkovski** [comment]
+1

**Comment by webjoko** [comment]
> @zjoko the code snippet above makes the templates translatable and it works well with WPML. I know it's not the same as having it built into oxygen and it requires adding the code somehow, but it works.

Thank you. I appreciate that.

**Comment by marcorubiol** [comment]
I'm adding the code above to wordpress and I can see the strings in WPML string translation but is not render in the frontend.

Someone know what I'0m doing wrong

btw, @d-packs thank you for the code

**Comment by lindebjerg** [comment]
+1 vote from me. I have used WPML on several Sites since more than 10 years. For me the only Plugin to setup multilingual sites. I would love to see WPML work perfect with Oxygen, for me The Ultimate Visual Site Builder for WordPress!

**Comment by ewokee** [comment]
+1

**Comment by Sabaidi59** [comment]
+1

**Comment by Jehu** [comment]
Please take a look in this topic with WPML team. i18n is such an important issue for many websites.

Since Version 4 (JSON instead of Shortcode) tranlastion via import of XLIFF is no more working because just the shortcode is updated on import. We have to stay on Oxygen 3.9 until this is fixed.

Oxygen people - please hear the need of your community.

**Comment by followleohans** [comment]
+1

**Comment by sergiobellini** [comment]
+1

**Comment by sergiobellini** [comment]
+1

**Comment by DeveloperCorsaro** [comment]
+1

**Comment by birralight13** [comment]
+1

**Comment by nuriarita** [comment]
+1

**Comment by little-gatsby** [comment]
+1

**Comment by grldf** [comment]
+1

**Comment by pad52** [comment]
+1

**Comment by uhlhosting** [comment]
Any ETA on this matter? Would love to see wpml support in oxygen.

**Comment by Vladulescu** [comment]
+1

**Comment by davidperroud** [comment]
+1 Support Oxygen for WPML Plugin ! 
Maybe go with the Go Global Programm of WPML ? 

I worked for 2 years on a multilingual website, with Oxygen and WPML, and everything was working fine before the version 4. 
Know i have to "hack" the Oxygen plugin if i want my visitors to get access to the website in all languages...

**Comment by LEloffel** [comment]
+1 for WPML support! Please

**Comment by CallanAH** [comment]
+1

**Comment by sercanvirlan** [comment]
+1

**Comment by wieseljonas** [comment]
+1

**Comment by fily1212** [comment]
+1, please!

**Comment by visionlabit** [comment]
+1 Support Oxygen for WPML Plugin Please!

**Comment by Alex-Rubio** [comment]
+1 please! WPML so useful

**Comment by mnwalker** [comment]
+1

**Comment by jorgetbruno** [comment]
+1

**Comment by uhlhosting** [comment]
@Spellhammer why this lack of reply from Oxygen? WPML says they are in real need for your collaboration to solve these issues. We are dealing here with years of working with WPML, LTD license that served us well on all websites without oxygen. This is a point where Elementor and other competition is doing steps while Oxygen lacks, either you folks don't know how to say f**k off since silence is not golden, ignorance is not bliss. Funny that you worked integrations that are not even so needed, yet the fact that oxygen websites cannot be translated properly with one of oldest, and most used translation plugins does not seem to bother anyone in this team. Such a shame having people to +1 + +1+1+11+1111+11111 here without any results from your side. Over 2 years now and wind is blowing. I start loosing trust that you really care about the customers in any way.

**Comment by lowcut** [comment]
+1

**Comment by omicronn** [comment]
+1

**Comment by TSMadmin** [comment]
+1

**Comment by ediger86** [comment]
+1. This is badly needed

**Comment by Globoris** [comment]
+1 @ediger86 is right: This is badly needed!

**Comment by jtlg28** [comment]
+1

**Comment by psomby** [comment]
WPML support would be very much appreciated!!

**Comment by mark88008** [comment]
How is it progressing with WPML support for Oxygen?

**Comment by TSMadmin** [comment]
Currently we are not using Oxygen with WPML because there is no support.

From: Michael Blechinger ***@***.***>
Date: Monday, January 9, 2023 at 5:51 AM
To: soflyy/oxygen-bugs-and-features ***@***.***>
Cc: Admin Accounts List ***@***.***>, Comment ***@***.***>
Subject: Re: [soflyy/oxygen-bugs-and-features] Support for WPML plugin (#938)

How is it progressing with WPML support for Oxygen?

**Comment by V0IDNIL** [comment]
Me, I act under the assumption that support for WPML will never come to Oxygen. I would be pleasantly surprised if it happens, but hope I have absolutely none. 

I strongly dislike WPML. It’s a sluggish, bloated mess of a plugin and their “support” is very much hit and miss. I’d like to use Weglot (which Louis Reingold publicly voiced support for) for each of my WordPress projects but have to use WPML (sometimes Polylang) most of the time. I’m based in Japan and practically each website I make is at least bilingual (Japanese/English), often three or more languages (usually Mandarin Chinese, Korean, Thai …). 
Weglot’s approach is to have identical site structures for each language (I know there’s a hack to hide content through css selectors, but that’s a hack, nothing workable for a client). That approach works fine if you have a simple website and/or your languages are culturally relatively similar (like European languages). In my case, practically all my Japanese clients (close to a hundred over the past five years) demand a different site structure for their Japanese visitors and their international visitors. Sometimes it’s just one extra page, often it’s much more. Not even to speak of those cases that have some e-commerce solution, in which case Weglot is completely unusable. That’s why I need to use something like WPML because it allows me to build independent websites for each language.

And that’s why I’ve turned my back on Oxygen and use another builder that natively supports WPML. I think you should all evaluate if you really need WPML or if you can instead use some alternative like WeGlot . And those of you who really need WPML, I recommend giving up all hope on Oxygen and use some other builder, and just start building your multilingual websites. Oxygen is fantastic when it works, but the current state of Oxygen+WPML is the stuff of nightmares.

**Comment by kotkotan** [comment]
Vote to fix WPML for Oxygen!

**Comment by MarcLob** [comment]
I got unfortunately mislead by a post on WPML about it's compatibility two years ago and got Oxygen because of this.

I don't think that Oxygen team will be doing must about this after all this time of inaction.

I do not suggest to buy this page builder if you need 100% translations functionality.
So if you are checking before you buy, please check the first link below and make your decision after.
They are some work around but there is other issues that come with this when updating site content particularly.

I did get good support form WPML as they tried hard to find some solutions.

Just got an email today from WPML about this and almost 2 years after posting on this support issue...
-------

We're sorry but we could not get cooperation from the Oxygen developers on this issue. 

We mention the following here.

https://wpml.org/documentation/plugins-compatibility/oxygen-visual-builder/

Please note that Oxygen Builder is currently not compatible with WPML. We reached out to their team to fix the current issues related to translating the content built using Oxygen.

If you need to use Oxygen with WPML, we suggest reaching out to their team and asking them to work with us on fixing the current issues. One of the ways of doing this is by upvoting and commenting on the open issue on the support page they already have for this.

https://github.com/soflyy/oxygen-bugs-and-features/issues/938
-------

> Voting for full compatibility ASAP with WPML.
> 
> I am from the DIVI ( Bloated) world and am looking forward to have a good page builder like Oxygen compatible with WPML.
> 
> **For does looking for some help on the subject:**
> 
> * https://wpml.org/forums/topic/oxygen-builder-translation/#post-8434339
> * https://wpml.org/documentation/plugins-compatibility/oxygen-visual-builder/#translating-oxygen-reusable-elements-templates
> 
> For WPML config file to get some fields into translation:
> 
> `<wpml-config>` `<shortcodes>` `<shortcode>` ` <tag>oxy_rich_text</tag>` ` </shortcode>` ` <shortcode>` ` <tag>ct_link_text</tag>` `</shortcode>` ` <shortcode>` ` <tag>ct_link_button</tag>` ` </shortcode>` ` </shortcodes>` `</wpml-config>`
> 
> IF any body knows how to get the link to translation in any modules with links that would help! Maybe a attribute tag?
> 
> Thanks!

I got unfortunately mislead by a post on WPML about it's compatibility two years ago and got Oxygen because of this.

I don't think that Oxygen team will be doing must about this after all this time of inaction.

I do not suggest to buy this page builder if you need 100% translations functionality.
So if you are checking before you buy, please check the first link below and make your decision after.
They are some work around but there is other issues that come with this when updating site content particularly.

I did get good support form WPML as they tried hard to find some solutions.

Just got an email today from WPML about this and almost 2 years after posting on this support issue...
-------

We're sorry but we could not get cooperation from the Oxygen developers on this issue. 

We mention the following here.

https://wpml.org/documentation/plugins-compatibility/oxygen-visual-builder/

Please note that Oxygen Builder is currently not compatible with WPML. We reached out to their team to fix the current issues related to translating the content built using Oxygen.

If you need to use Oxygen with WPML, we suggest reaching out to their team and asking them to work with us on fixing the current issues. One of the ways of doing this is by upvoting and commenting on the open issue on the support page they already have for this.

https://github.com/soflyy/oxygen-bugs-and-features/issues/938
-------

**Comment by followleohans** [comment]
Having to choose between WPML and Oxygen, I choose the later one.
Oxygen builder is the best builder with amazing SEO results. 
WPML is buggy regardless of the builder.

**Comment by netrunner7** [comment]
+1 for better polylang or any WPML

**Comment by mark88008** [comment]
1) WPML, polylang and translatepress
2) Then fully fixed drag and drop for the structure panel on the right sidebar and the canvas

**Comment by vereinteplaneten** [comment]
+1 for WPML and polylang

**Comment by Antoinevenco** [comment]
+1 for WPML and polylang

**Comment by leodedro** [comment]
+1 for WPML and Polylang support.
Oxygen is an amazing tool, but at the same time Multilanguage is crucial in many cases.

**Comment by KristapsVilcans** [comment]
WPML, we need this support for Oxygen as Oxygen is stronger and better every day.

**Comment by andreasoikon** [comment]
+1 Support Oxygen for WPML plugin!

**Comment by Ikpixel** [comment]
+1 as you know WPML wants to work with you to solve the problems.

**Comment by jevgen** [comment]
+1

**Comment by Qbaska** [comment]
This is ridiculous, 2 years and no official statement... Why cant Oxygen team just say oficialy "FU** OFF people, we will never support WPML"? This would save time and money for hundreds of people. This is shady as f***. Instead Oxygen developers are shifting focus towards Breakdance, which im sure is castrated crap on subscription plan.

By the way, WooCommerce pages are not translatable at all, i just had super long conversation with WPML supporter on their chat. Super nice guy was extremely determined to help me but ultimately we figured out that its absolutely not possible to do multilingual WooCommerce page with WPML and Oxygen. Other templates might work but not the most important ones for online store...  Also he said its Oxygen team who does not want to solve this issue!! WPML team tries to reach them but with no luck.

For anyone that reads this: ITS NOT POSSIBLE TO BUILD WOOCOMMERCE PAGE with Oxy and WPML All fixes and workarounds does not apply to WooCommerce-related templates. Trust me - I tried them all, My fixes were also checked by WPML team member and still no luck. DO NOT BUY OXYGEN IF YOU WANT TO BUILD ONLINE STORE!! DO NOT SPEND MONEY ON OXYGEN IF YOU NEED WPML!! Use something else... Elementor?

**Comment by uhlhosting** [comment]
@Qbaska it is sad at same time, to ignore the demand of so many users.

**Comment by Spellhammer** [comment]
Anyone who has had trouble with WPML and Oxygen should try 4.6 RC 1 from https://oxygenbuilder.com/portal.

I just translated a WooCommerce product/shop with no trouble at all: https://www.loom.com/share/748d80f59d2744f38d604f656b221490

There may still be edge cases, but compatibility is vastly improved in 4.6 RC 1.

**Comment by ewokee** [comment]
> Anyone who has had trouble with WPML and Oxygen should try 4.6 RC 1 from https://oxygenbuilder.com/portal.
> 
> I just translated a WooCommerce product/shop with no trouble at all: https://www.loom.com/share/748d80f59d2744f38d604f656b221490
> 
> There may still be edge cases, but compatibility is vastly improved in 4.6 RC 1.

haha so its not there but its there... crazy and funny. and fits very well to the whole oxygen situation. i really wish all good luck.

**Comment by followleohans** [comment]
Is it possible to translate buttons now?

**Comment by ewokee** [comment]
another fun fact. they now suggest to use a RC version. which still don't even offer an official support for wpml....

**Comment by uhlhosting** [comment]
> another fun fact. they now suggest to use a RC version. which still don't even offer an official support for wpml....

I believe what @Spellhammer was suggesting is that things slowly are moving towards better compatibility. So lets give this a chance. Im sure in less than a month or so the beta will be public. So we can test it meanwhile. No one said to use it in production.

**Comment by followleohans** [comment]
> Oxygen RC versions are stable enough for use in production.

@Spellhammer is it possible to translate buttons now?

**Comment by jevgen** [comment]
> > Oxygen RC versions are stable enough for use in production.
> 
> @Spellhammer is it possible to translate buttons now?

He clearly showed in the video on Loom, that its in WooC. You may try to find button text in String Translation in WPML for WooC and translate it there.

**Comment by uhlhosting** [comment]
> Oxygen RC versions are stable enough for use in production.

@Spellhammer does it update now the translation of pages properly?

**Comment by kapaweb** [comment]
I want to use WPML in a website made with OXYGEN. It is written [HERE ](https://oxygenbuilder.com/documentation/other/multilingual-with-wpml/) that OXYGEM is compatible with the WPML. 
Please solve all issues soon, because this creates serious problems in our work.
Thank you.

**Comment by Tom-H-L** [comment]
What is the current state concerning Oxygen and multi-lingual websites?

The European Union has 24 official languages in 27 member states that resemble together a unified market of 446 million people (USA: 333M). Anyone serious about doing business and commerce throughout this huge market absolutely depends on multi-language support of his website builder.

This feature request is 3 years and 1 month old. Over 100 users have upvoted it. Until recently people write about it in a way that it seems that the feature still is not in place.

On the Oxygen [website](https://oxygenbuilder.com/documentation/other/multilingual-with-wpml/) I can read this:
"Oxygen has rudimentary support for WPML, Polylang and Weglot", "Note that the Oxygen and WPML teams have been in contact and are jointly investigating official compatibility."

On the other hand, on the WPML [website](https://wpml.org/documentation/plugins-compatibility/oxygen-visual-builder/) they write:
"Please note that Oxygen Builder is currently not compatible with WPML. We reached out to their team to fix the current issues related to translating the content built using Oxygen.

If you need to use Oxygen with WPML, we suggest reaching out to their team and asking them to work with us on fixing the current issues. One of the ways of doing this is by [upvoting and commenting on the open issue](https://github.com/soflyy/oxygen-bugs-and-features/issues/938) on the support page they already have for this."

So it seems that the WPML team tried to make it happen together with the Oxygen team, but Oxygen denied to work together with them?

@Spellhammer please provide transparency about what the plans of Oxygen are concerning supporting WPML or any other multi-language support and especially what the current state of implementation is, if any, including an outlook for an ETA for this feature.

Thanks for any information on this!

**Comment by omicronn** [comment]
+1 @Tom-H-L

**Comment by Mamadzhanov** [comment]
The only thing that stops our agency to switch to Oxygen is lack of support of multilanguage feature. As 90% of our clients demand websites to be available in German and English, the WPML plugin is a must, thereby we have purchased the most expensive license.
Sad that for a few years, there is no big update from Oxygen team, we have to stick to other tools and builders.
We seriously need a multi-language support (WPML).

**Comment by uhlhosting** [comment]
@Mamadzhanov the level of ignorance practices here, is one of high skills. At first the founder of Oxygen, made a promise that WPML is supported, in a youtube video, later few years we all know is working yet with very many downsides, and that promise turned into dust. 

Now they made a fix no one knows what exactly, since that was reverted. So in fact they made more dust. No real changes in WPML + Oxygen for years.

**Comment by louisreingold** [comment]
@uhlhosting 

We stand by our comments in this thread and what is written in the documentation at https://oxygenbuilder.com/documentation/other/multilingual-with-wpml/.

The video you link to is clearly says "beta" and was also for Oxygen 1.x, which was completely discontinued.

**Comment by Tom-H-L** [comment]
@louisreingold Thanks for joining the discussion.

Can you please give guidance / informormation on this, so that we can estimate how good Oxygen would work for multi-linguage sites:

- https://github.com/soflyy/oxygen-bugs-and-features/issues/938#issuecomment-1536436958
- https://github.com/soflyy/oxygen-bugs-and-features/issues/938#issuecomment-1536223373

**Comment by spinoWeb2** [comment]
> Translatepress makes it seem [here](https://translatepress.com/how-to-translate/oxygen-wordpress-sites-translatepress/) like it works 100% out of the box with Oxygen. Can someone confirm this? I can't find any information on the Oxygen website about this. Or suggest any other multi-language plugin that works with Oxygen?

@Tom-H-L 
I have regularly used Oxygen Builder and Translatepress together on dozens of sites over the past 2 years.
Everything can be translated. Maybe sometimes you need to do some editing in the original blocks to get them recognized. For example, changing form type (from text to rich-text and vice versa) or packing elements into divs or spans...
In the end, however, everything can be translated.

Rather I found some issues between Translatepress and other plugins, like CPT UI archives, CF7 email, search and filter plugins...

Having a frontend approach the translation is quick and easy, but the management is lacking compared to WPML. But the comparison between the 2 is another matter...

**Comment by followleohans** [comment]
With WPML you can't translate buttons created in Oxygen Builder.  That makes it useless.

**Comment by Tom-H-L** [comment]
> @Tom-H-L I have regularly used Oxygen Builder and Translatepress together on dozens of sites over the past 2 years. Everything can be translated. Maybe sometimes you need to do some editing in the original blocks to get them recognized. For example, changing form type (from text to rich-text and vice versa) or packing elements into divs or spans... In the end, however, everything can be translated.

Thank you for that information, much appreciated!
So as it seems, due to Translatepress, someone CAN use Oxygen for multilingual websites. This brings Oxigen for us back in the game, we had eliminated it from the short list of possible solutions already due to the suspected non-compatibility with any multilingual plugin.

> Having a frontend approach the translation is quick and easy, but the management is lacking compared to WPML. But the comparison between the 2 is another matter...

I thought of this, but since we need Wordpress only for small portfolio sites, I guess that the frontend approach remains maintainable, now?
We need this sites to be multilingual and SEO optimized.  Translatepress works fine for SEO, too, no?

**Comment by pad52** [comment]
Since I don't want to copy/paste every text field from a page to another inside the Oxygen Builder,
I just made a very simple python script to automate at least the translation process.
(you will keep doing copy paste between the JSON of a page to the JSON of the translated page)
Hope it helps someone.
https://gist.github.com/pad52/06938927329125a08ff4af474e4a60f2

**Comment by pad52** [comment]
I developed some tools for partially automate the translation process of oxygen websites with wpml.
Check it out here https://github.com/pad52/oxygen-wpml-tools

**Comment by uhlhosting** [comment]
@pad52 how you handle the conditions to display the correct language on language switch?

**Comment by pad52** [comment]
I am translating in a similar way of @soloconfusione (mentioned here: https://github.com/soflyy/oxygen-bugs-and-features/issues/938#issuecomment-1710465734) 

So I am translating templates and pages using the method (and switch) provided by WPML.

The only difference is that instead of translate manually every page or template I am using my tools to generate the JSONs out of the original JSON and a .PO file, then I paste the result in the JSON tab of the translated page/template.

I am not using the Oxygen conditions method.

**Comment by uhlhosting** [comment]
@pad52 not sure I get it how would that work without the conditions, if thats possible, ohmy, lifesaver. Can you make a short video maybe on the whole process? I am sure many can benefit from it.

**Comment by pad52** [comment]
1. Enable the translation of the templates in WPML settings (I enable everything that is Oxygen related since I don't know what all the elements means)
![Screenshot_20231005_122034](https://github.com/soflyy/oxygen-bugs-and-features/assets/17023506/1f13d4cc-1de3-4a68-bec4-b85842152d84)

2. Go in the template section and add a translation for the template exaclty in the same way that you do on normal pages.
![Screenshot_20231005_122338](https://github.com/soflyy/oxygen-bugs-and-features/assets/17023506/97b90f8f-4e78-44fb-9972-2f016b274406)

3. Translate the template via Oxygen Builder or copy/pasting json: be careful inthe "Where does this template apply" chechboxes - they will not get duplicated - you need to check them again. 
![Screenshot_20231005_122406](https://github.com/soflyy/oxygen-bugs-and-features/assets/17023506/a30ebd6a-592e-4015-8239-2b76e30225d9)

For pages everything is the same except "Where does this template apply" part.

**Comment by pad52** [comment]
Unfortunately, not working anymore...

- Now using per-page & per-product template setting
(IT-single_product_page, EN-single_product_page to be assigned to single products &
IT-contact_page, EN-contact_page to be assigned to let's say "contact" page) 

- Gettext php functions in code block for texts in single template archive pages (BLOG, BLOG post, 404, Search..)
 
I tried but cannot make conditions working...

**Comment by uhlhosting** [comment]
@pad52 I got the very same issue, reported it to Oxygen team and @Spellhammer ensured me is not a Oxygen issue and recommended me to DISABLE WPML AND TRY AGAIN, hats of of course I toke it as a joKE, since one cannot expect to DEBUG a problem between 2 plugins here OXYGEN / WPML without one of them active. So Oxygen team amazes me with their intelligence day by day. 

I say like this, we need to wait until wordpress ships native translations that will be soon! And if even then Oxygen will fail to bring it up to the expections of their users, well, they can vanish from the field easy! They have no clue, they think they have the world at their feet, we are that world, we are the ones who paid for something that was HALF fake promises and half lies, not to mention when the very CEO of the company have a video in version 2 of oxygen, saying how they work hard to make better integration between WPML and Oxygen, yeah, keep on dreaming guys.

Tbh lately me personally I stopped opening support requests, when I seen their lack of care and interest to solve THE VERY core issues that Elijah, refused to see and accept as they are, just because we use WPML. HOW back minded and poor minded that mentality is? Well I let you guys decide. Be carefully with such people who promise the seas and mountains, and you end up with a pool and a rock...

------

Shame on you all OXYGEN core developers! Shame me once, yet now is time to say, shame on you twice! And 3 times until you will listen our voice, and move from your sleeping beauty and comfort zone, and start solving solely REAL issues we all know they are too obvious to IGNORE!



Happy new year! And hopefully we will eventually get a chance to stick to Oxygen, the idea of some people behind the works there are spectacular, the other few moves of people with poor thinking before acting, that move we all remember where the CEO decided to pay shitload of cash to get into competition to Elementor, whereas a smart person, would had refurbished Oxygen into a product deem to beat both Elementor and or any other competition. So he dismantled best devs @ oxygen, since there is proof they almost all left, and those who stayed they shifted focus on other directions. 

Do they learned from that? Not really, the only they learned is that when you disrespect the suggestions of an entire community you pay the price of loosing that community. Aren't we seeing here a massive drop in new leads onboarding Oxygen? 

Ohh yes we do, bricks gain space in all areas where Oxygen is not anymore. Good for them, only time will say when every chapter will end, hence all that starts will end one day!

**Comment by uhlhosting** [comment]
93 People participating in this thread only! All our voices nulled and ignored, what a great customer care and support! Prima!

**Comment by kademario** [comment]
Hi everyone,

I ran into the same Oxygen + WPML workflow problem while working on multilingual Oxygen sites.

We ended up building an independent connector for this specific use case: **WPML Oxygen Connector**.

The idea is to avoid rebuilding or manually duplicating Oxygen layouts for each language. The plugin reads Oxygen builder data, detects translatable content, registers it with WPML String Translation, and renders the translated content on the frontend while keeping the Oxygen layout structure intact.

It currently supports Oxygen Builder 6 and Oxygen 4 Classic, and includes:

- scanning Oxygen pages/templates/blocks,
- inline translation inside WordPress,
- JSON export/import for external or AI-assisted translation,
- frontend rendering of translated Oxygen content,
- cache regeneration and layout synchronization.

It is an independent third-party plugin, not an official WPML or Oxygen integration, but it may help people who are blocked by the current Oxygen + WPML limitations.

Product page:
https://www.kademar.io/en/wpml-oxygen-connector

Documentation:
https://www.kademar.io/en/documentation-wpml-oxygen-connector/

Hope this helps anyone maintaining multilingual Oxygen websites.
